Unit4Assignment1Proj/Unit4Assignment1Proj/Views/AddPhotoViewController.swift

```diff
@@ -0,0 +1,166 @@
+//
+//  AddPhotoViewController.swift
+//  Unit4Assignment1Proj
+//
+//  Created by Kevin Natera on 10/6/19.
+//  Copyright © 2019 Kevin Natera. All rights reserved.
+//
+
+import UIKit
+import Photos
+
+class AddPhotoViewController: UIViewController {
+
+    @IBOutlet weak var imageOutlet: UIImageView!
+    @IBOutlet weak var titleTextView: UITextView!
+    @IBOutlet weak var captionTextView: UITextView!
+    @IBOutlet weak var cancelButtonOutlet: UIBarButtonItem!
+    @IBOutlet weak var saveButtonOutlet: UIBarButtonItem!
+    @IBOutlet weak var photoLibraryButtonOutlet: UIBarButtonItem!
+    @IBOutlet weak var camerabuttonOutlet: UIBarButtonItem!
+    @IBOutlet weak var titleLabel: UILabel!
+    @IBOutlet weak var captionLabel: UILabel!
+    
+    
+    weak var delegate : AddPhotoDelegate?
+    
+    var darkModeEnabled = false
+    
+    @IBAction func cancelButtonPressed(_ sender: UIBarButtonItem) {
+        self.dismiss(animated: true, completion: nil)
+    }
+    
+    @IBAction func saveButtonPressed(_ sender: UIBarButtonItem) {
+        photoTitle = titleTextView.text
+        caption = captionTextView.text
+        
+        if photoTitle != "" && imageName != "" && photoToBeEdited == nil {
+            let newPhoto = Photo(title: photoTitle, caption: caption, imageName: imageName, date: Date())
+                DispatchQueue.global(qos: .utility).async {
+                    try? PhotoPersistenceHelper.manager.save(newPhoto: newPhoto)
+                }
```

Comment:
In the "else" of this if statement you can add an alert letting the user know that in order to save to a journal they need to fill in the blacks